src/include/distributed/metadata_cache.h

```diff
@@ -117,6 +117,22 @@ typedef struct DistObjectCacheEntry
 } DistObjectCacheEntry;
 
 
+/*
+ * Methods to check if given partitionMethod and replicationModel would identify
+ * a citus local table or a reference table especially when the cache entry for
+ * table is not inserted yet.
+ */
+#define IsReferenceTableByParameters(partitionMethod, replicationModel) \
```

Comment:
no specific reason, we could also prefer to implement them in simple functions, doesn't much matter for me

Reply:
I'd go with both of them as functions because It seems a bit inconsistent that we use function for `IsCitusLocalTable` but a macro for `IsCitusLocalTableByDistParams`.